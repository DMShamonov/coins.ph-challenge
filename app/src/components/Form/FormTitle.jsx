// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { FormTitlePropsType } from './types.js.flow';

export default function FormTitle({
  className,
  children,
}: FormTitlePropsType): React.Node {
  return (
    <p className={classnames('form__title', className)}>{children}</p>
  );
}
