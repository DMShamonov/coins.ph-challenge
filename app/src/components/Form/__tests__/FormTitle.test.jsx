import React from 'react';
import { shallow } from 'enzyme';

import FormTitle from '../FormTitle';

describe('`FormTitle` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<FormTitle>__CONTENT__</FormTitle>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<FormTitle className="__CLASSNAME__">__CONTENT__</FormTitle>)).toMatchSnapshot();
  });
});
