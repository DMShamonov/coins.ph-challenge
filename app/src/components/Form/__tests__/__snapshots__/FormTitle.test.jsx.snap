// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`FormTitle\` component should render correctly with \`className\` prop 1`] = `
<p
  className="form__title __CLASSNAME__"
>
  __CONTENT__
</p>
`;

exports[`\`FormTitle\` component should render correctly with default props 1`] = `
<p
  className="form__title"
>
  __CONTENT__
</p>
`;
